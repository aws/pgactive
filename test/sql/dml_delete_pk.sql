-- RT #37826 "issuing a DELETE broken replication"
SELECT * FROM public.pgactive_regress_variables()
\gset

\c :writedb1

BEGIN;
RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($$
	CREATE TABLE public.test (
		id TEXT,
		ts TIMESTAMP DEFAULT ('now'::TEXT)::TIMESTAMP,
		PRIMARY KEY (id)
	);
$$);
COMMIT;

SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);

-- INSERT data
INSERT INTO test (id, ts) VALUES ('row', '1970-07-21 12:00:00');
INSERT INTO test (id) VALUES ('broken');
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c :readdb2
SELECT id FROM test ORDER BY ts;

-- DELETE one row by PK
\c :writedb2
DELETE FROM test WHERE id = 'row';
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
SELECT id FROM test ORDER BY ts;

\c :readdb1
SELECT id FROM test ORDER BY ts;

\c :writedb1
DELETE FROM test WHERE id = 'broken';
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
SELECT id FROM test ORDER BY ts;

\c :readdb2
SELECT id FROM test ORDER BY ts;

\c :writedb1
BEGIN;
RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($$DROP TABLE public.test;$$);
COMMIT;
