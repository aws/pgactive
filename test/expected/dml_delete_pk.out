-- RT #37826 "issuing a DELETE broken replication"
SELECT * FROM public.pgactive_regress_variables()
\gset
\c :writedb1
BEGIN;
RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($$
	CREATE TABLE public.test (
		id TEXT,
		ts TIMESTAMP DEFAULT ('now'::TEXT)::TIMESTAMP,
		PRIMARY KEY (id)
	);
$$);
 pgactive_replicate_ddl_command 
---------------------------
 
(1 row)

COMMIT;
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
----------------------------------------
 
(1 row)

-- INSERT data
INSERT INTO test (id, ts) VALUES ('row', '1970-07-21 12:00:00');
INSERT INTO test (id) VALUES ('broken');
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
----------------------------------------
 
(1 row)

\c :readdb2
SELECT id FROM test ORDER BY ts;
   id   
--------
 row
 broken
(2 rows)

-- DELETE one row by PK
\c :writedb2
DELETE FROM test WHERE id = 'row';
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
----------------------------------------
 
(1 row)

SELECT id FROM test ORDER BY ts;
   id   
--------
 broken
(1 row)

\c :readdb1
SELECT id FROM test ORDER BY ts;
   id   
--------
 broken
(1 row)

\c :writedb1
DELETE FROM test WHERE id = 'broken';
SELECT pgactive.pgactive_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 pgactive_wait_for_slots_confirmed_flush_lsn 
----------------------------------------
 
(1 row)

SELECT id FROM test ORDER BY ts;
 id 
----
(0 rows)

\c :readdb2
SELECT id FROM test ORDER BY ts;
 id 
----
(0 rows)

\c :writedb1
BEGIN;
RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($$DROP TABLE public.test;$$);
 pgactive_replicate_ddl_command 
---------------------------
 
(1 row)

COMMIT;
